internal/commands/build/types.go

```diff
@@ -0,0 +1,73 @@
+package build
+
+import (
+	"fmt"
+	"reflect"
+
+	"github.com/docker/cli/cli/compose/loader"
+	compose "github.com/docker/cli/cli/compose/types"
+	"github.com/pkg/errors"
+)
+
+// A minimal subset of github.com/docker/cli/cli/compose/types/types.go for the purpose of loading the build configuration
+
+type ServiceConfig struct {
+	Name string `yaml:"-" json:"-"`
+
+	Build *ImageBuildConfig
+	Image *string
+}
+
+type ImageBuildConfig struct {
+	Context    string                    `yaml:",omitempty" json:"context,omitempty"`
+	Dockerfile string                    `yaml:",omitempty" json:"dockerfile,omitempty"`
+	Args       compose.MappingWithEquals `yaml:",omitempty" json:"args,omitempty"`
+}
+
+func Load(dict map[string]interface{}) ([]ServiceConfig, error) {
+	section, ok := dict["services"]
+	if !ok {
+		return nil, fmt.Errorf("compose file doesn't declare any service")
+	}
+	services, ok := section.(map[string]interface{})
+	if !ok {
+		return nil, fmt.Errorf("Invalid compose file: 'services' should be a map")
+	}
+	return LoadServices(services)
+}
+
+func LoadServices(servicesDict map[string]interface{}) ([]ServiceConfig, error) {
```

Comment:
nit: Should this function be exported?

Reply:
not strictly required indeed. Copy/Pase from cli/compose/loader